widgets/star-rating/star-rating.js

```diff
@@ -0,0 +1,165 @@
+
+  let RefinementList = headerFooterHOC(require('../../components/RefinementList/RefinementList.js'));
+  if (autoHideContainer === true) {
+    RefinementList = autoHideContainerHOC(RefinementList);
+  }
+
+  if (!container || !facetName) {
+    throw new Error(usage);
+  }
+
+  return {
+    getConfiguration: () => {
+      return {
+        disjunctiveFacets: [facetName]
+      };
+    },
+
+    render: function({helper, results, templatesConfig, state, createURL}) {
+      let templateProps = utils.prepareTemplateProps({
+        transformData,
+        defaultTemplates,
+        templatesConfig,
+        templates
+      });
+
+      let facetValues = [];
+      let allValues = {};
+      for (let v = max - 1; v >= 0; --v) {
+        allValues[v] = 0;
+      }
+      results.getFacetValues(facetName).forEach(facet => {
+        let val = Math.round(facet.name);
+        if (!val || val > max - 1) {
+          return;
+        }
+        allValues[val] += facet.count;
+      });
+      let refinedStar = this._getRefinedStar(helper);
+      for (let star = max - 1; star >= 1; --star) {
+        let count = allValues[star];
+        if (refinedStar && star !== refinedStar && count === 0) {
+          // skip count==0 when at least 1 refinement is enabled
+          continue;
+        }
+        let stars = [];
+        for (let i = 1; i <= max; ++i) {
+          stars.push(i <= star);
+        }
+        facetValues.push({
+          stars: stars,
+          name: '' + star,
+          count: count,
+          isRefined: refinedStar === star,
+          labels
+        });
+      }
+
+      let cssClasses = {
+        root: cx(bem(null), userCssClasses.root),
+        header: cx(bem('header'), userCssClasses.header),
+        body: cx(bem('body'), userCssClasses.body),
+        footer: cx(bem('footer'), userCssClasses.footer),
+        list: cx(bem('list'), userCssClasses.list),
+        item: cx(bem('item'), userCssClasses.item),
+        link: cx(bem('link'), userCssClasses.link),
+        disabledLink: cx(bem('link', 'disabled'), userCssClasses.disabledLink),
+        count: cx(bem('count'), userCssClasses.count),
+        star: cx(bem('star'), userCssClasses.star),
+        emptyStar: cx(bem('star', 'empty'), userCssClasses.emptyStar),
+        active: cx(bem('item', 'active'), userCssClasses.active)
+      };
+
+      ReactDOM.render(
+        <RefinementList
+          createURL={(stars) => {
+            return createURL(state.toggleRefinement(facetName, stars));
+          }}
+          cssClasses={cssClasses}
+          facetValues={facetValues}
+          shouldAutoHideContainer={results.nbHits === 0}
+          templateProps={templateProps}
+          toggleRefinement={this._toggleRefinement.bind(this, helper)}
+        />,
+        containerNode
+      );
+    },
+
+    _toggleRefinement: function(helper, facetValue) {
```

Comment:
Love this notation we should use it in other widgets rather than externalized complex fns